### PR TITLE
Add hometask 17

### PR DESCRIPTION
**Pull request by ZlataKutuzava**

### CODE REVIEW

File: hw17/spec/calc.spec.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
+import { Calculator } from "../src/calculator";
+import assert from "assert";
+const calc = new Calculator();
+
+describe("Unit tests for the Calculator - Mocha", () => {
+  it("The addition of 4 and 3 is 7", () => {
```
Comment (VladSelikh) [inline hw17/spec/calc.spec.ts:6]: Please don't use hardcode in the tests names.
In this very case you can write:
`Should correctly add two positive numbers`.
The same is true for all your other tests.

File: babel.config.js [hunk 2]
```diff
@@ -0,0 +1,2 @@
+export const presets = ["@babel/preset-env", "@babel/preset-react", "@babel/preset-typescript"];
+export const plugins = ["@babel/plugin-transform-runtime"];
```
Comment (VladSelikh) [inline babel.config.js:2]: What do you need this `babel` config file for?

File: package.json [hunk 3]
```diff
@@ -5,7 +5,9 @@
   "main": "index.js",
   "scripts": {
     "hw16": "npx ts-node ./hw16/src/index.ts",
-    "test": "echo \"Error: no test specified\" && exit 1"
+    "hw17-test-mocha": "npx mocha -r ts-node/register ./hw17/spec/calc.spec.ts",
```
Comment (VladSelikh) [inline package.json:8]: Please rerun. This command results in a TypeScript error.

File: package.json [hunk 4]
```diff
@@ -5,7 +5,9 @@
...
+    "hw17-test-jest": "npx jest ts-node/register ./hw17/spec/jest_calc.spec.ts",
```
Comment (VladSelikh) [inline package.json:9]: Please point at the config file, not the spec file.

File: jest.config.cjs [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+      "<rootDir>/node_modules/(?!nanostores)"
+    ],
+    "globals": {
+      "ts-jest": {
+        "tsconfig": {
+          "allowJs": true
+        }
+      }
+    },
+  };
```
Comment (VladSelikh) [inline jest.config.cjs:19]: Please place this config file in the corresponding directory. In your case this is a `hw17` folder.

**Comment by VladSelikh** [comment]
I don't see any `tsconfig` file. Please add it.

File: babel.config.js (hunk 2 above)
Comment (ZlataKutuzava) [reply babel.config.js:2]: Removed

File: hw17/spec/calc.spec.ts (hunk 1 above)
Comment (ZlataKutuzava) [reply hw17/spec/calc.spec.ts:6]: Updated

Comment (ZlataKutuzava) [reply hw17/spec/calc.spec.ts:6]: Updated

File: babel.config.js (hunk 2 above)
Comment (ZlataKutuzava) [reply babel.config.js:2]: Removed

File: jest.config.cjs (hunk 5 above)
Comment (ZlataKutuzava) [reply jest.config.cjs:19]: Done. Added config files to mocha and jest

File: package.json (hunk 4 above)
Comment (ZlataKutuzava) [reply package.json:9]: Done

File: package.json (hunk 3 above)
Comment (ZlataKutuzava) [reply package.json:8]: Rerun. Tests are executed without TypeScript error.

File: hw17/jest.config.cjs [hunk 6]
```diff
@@ -0,0 +1,9 @@
+/** @type {import('ts-jest/dist/types').InitialOptionsTsJest} */
+// eslint-disable-next-line no-undef
+module.exports = {
+    preset: 'ts-jest',
+    testEnvironment: 'node',
+    testMatch: ['**/hw17/jest-test/**/*.spec.ts'],
```
Comment (VladSelikh) [inline hw17/jest.config.cjs:6]: Please delete all unnecessary linebreaks.

File: package.json [hunk 7]
```diff
@@ -18,10 +19,18 @@
   },
   "homepage": "https://github.com/ZlataKutuzava/tms#readme",
   "dependencies": {
+    "@babel/preset-typescript": "^7.18.6",
```
Comment (VladSelikh) [inline package.json:22]: What do yiu need this package for?

File: package.json [hunk 8]
```diff
@@ -18,15 +19,24 @@
   },
   "homepage": "https://github.com/ZlataKutuzava/tms#readme",
   "dependencies": {
+    "@types/jest": "^28.1.6",
+    "@types/mochawesome": "^6.2.1",
+    "@types/superagent": "^4.1.15",
+    "chai": "^4.3.6",
+    "jest": "^28.1.3",
+    "mochawesome": "^7.1.3",
+    "superagent": "^8.0.0",
```
Comment (VladSelikh) [inline package.json:28]: What do you need `superagent` and `@types/superagent` packages now? You don't use them in your homework.

**Review by VladSelikh — APPROVED** [review]

**Pull request merged**